docs/OV_Runtime_UG/ov_dynamic_shapes.md

```diff
@@ -0,0 +1,110 @@
+as well as increased amount of consumed memory.
+As this method cannot be scaled well it is used in combination with the padding:
+model with the most suitable input shape among pre-reshaped models is chosen.
+It gives smaller pad area in comparison to a single model.
+
+Let's mention another practical but still complicated approach when input tensor can be divided into multiple chunks along dynamic dimension.
+For example, if we have a batch of independent inputs as a single tensor.
+If arbitrary division along batch dimension is really possible -- and for batch dimension it should be possible by the dimension purpose --
+we can run multiple inferences using approach with several pre-compiled models choosing sized to have minimal number of inferences
+having a particular batch size in input tensor.
+
+Say if there are models pre-compiled for batch sizes 1, 2, 4 and 8,
+input tensor with batch 5 can be processed with two infer calls with batch size 1 and 4.
+(Here we assume the batch processing is required for performance reasons otherwise we can just loop over images in a batch,
+and process image by image with a single compiled model.)
+
+##Dynamic Shapes without Tricks
+
+In this section we describe how to handle dynamically shaped models more naturally with OpenVINO 2.0 API.
+To avoid the tricks described in the previous section there is a way to directly say that one or multiple dimensions in model input is going to be dynamic.
+This is achieved with the same reshape method that is used for alternating static shape of inputs.
+Dynamic dimensions are marked with `-1` or `ov::Dimension()`.
+Both means the same: undefined dimension.
+
+@snippet snippets/ov_dynamic_shapes.cpp ov_dynamic_shapes:reshape_undefined
+
+In this and following examples we assume that the model has a single input and single output that let us simplify the code.
+There are no restrictions on the number of inputs and output to apply dynamic shapes in the API though.
+
+Dynamic dimensions can exist in the input model without calling reshape.
+Many DL frameworks support undefined dimensions.
+If such a model is converted with Model Optimizer or read directly by Core::read_model, undefined dimensions are preserved.
+So we don't need to call reshape if undefined dimensions are already configured in a desired form in the original model or IR file.
+
+Besides marking a dimension just dynamic, we also can specify bounds for the dimension: lower bound and/or upper bound.
+This is a range of allowed values for the dimension.
+Bound are coded as arguments for `ov::Dimension`:
+
+@snippet snippets/ov_dynamic_shapes.cpp ov_dynamic_shapes:reshape_bounds
+
+Lower bound is not so important as upper bound, because knowing of upper bound allows plugins to more precisely allocate memory for intermediate tensors for inference and use lesser number of tuned kernels for different sizes.
```

Comment:
Consider restructuring the info about bounds:
- Lower bounds: what is it for, how to select optimal values
- Upper bounds: what is it for, how to select, limitations